client/lib/application/src/application-form/application-form.jsx

```diff
@@ -71,7 +71,19 @@ class ApplicationForm extends React.Component {
     /*eslint-disable react/no-direct-mutation-state */
     // TODO fix direct mutation of state here if possible
     update(field, prop, evt) {
-        if (field === 'criticality_level') {
+        
+        //Add an explicit warning if User unchecks the Active Application field
+        if (field === 'active') {
```

Comment:
There's a lot of code duplication with `this.state.app[field] = evt.target[prop];` + the if conditions are on 3 levels. Consider refactoring